python/ray/util/sgd/v2/session.py

```diff
@@ -0,0 +1,116 @@
+import queue
+import time
+import threading
+
+from ray.util.sgd.v2.constants import TIME_THIS_ITER_S
+
+
+class Session:
+    """Holds information for training on each worker."""
+
+    def __init__(self, training_thread: threading.Thread, world_rank: int):
```

Comment:
Should this just take in a `PropagatingThread` instead?

Reply:
Changed